subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/AbstractFingerprintChanges.java

```diff
@@ -16,18 +16,19 @@
 
 package org.gradle.internal.execution.history.changes;
 
-import com.google.common.collect.ImmutableSortedMap;
 import org.gradle.internal.change.ChangeContainer;
 import org.gradle.internal.change.ChangeVisitor;
 import org.gradle.internal.fingerprint.CurrentFileCollectionFingerprint;
 import org.gradle.internal.fingerprint.FileCollectionFingerprint;
 
+import java.util.SortedMap;
+
 public abstract class AbstractFingerprintChanges implements ChangeContainer {
-    protected final ImmutableSortedMap<String, FileCollectionFingerprint> previous;
-    protected final ImmutableSortedMap<String, CurrentFileCollectionFingerprint> current;
+    protected final SortedMap<String, FileCollectionFingerprint> previous;
+    protected final SortedMap<String, CurrentFileCollectionFingerprint> current;
```

Comment:
Why change this?

Reply:
We create two kinds of filtered changes from the inputs: the incremental one and the non-incremental one. For the non-incremental one, we only need to iterate once and don't need to query, so I figured we can use `Maps.filterKeys` to do the (lazy) filtering on the backing maps and save some cycles when we don't need it.

`Maps.filterKeys` only returns a `SortedMap`, which is sufficient for comparing the changes.